src/Commands/InitCommand.php

```diff
@@ -180,35 +183,35 @@ public function handle(): void
         }
 
         if ($shouldGenerateReadme = $this->confirm('Do you want to generate a README file?', true)) {
-            $this->fillReadme();
+            $this->readmeGenerator->generate($this->appName, $this->appType->value);
 
             if ($this->confirm('Do you need a `Resources & Contacts` part?', true)) {
-                $this->fillResourcesAndContacts();
+                $this->readmeGenerator->fillResourcesAndContacts();
                 $this->fillResources();
                 $this->fillContacts();
             }
 
             if ($this->confirm('Do you need a `Prerequisites` part?', true)) {
-                $this->fillPrerequisites();
+                $this->readmeGenerator->fillPrerequisites();
             }
 
             if ($this->confirm('Do you need a `Getting Started` part?', true)) {
                 $this->fillGettingStarted();
             }
 
             if ($this->confirm('Do you need an `Environments` part?', true)) {
-                $this->fillEnvironments();
+                $this->readmeGenerator->fillEnvironments($this->appUrl);
```

Comment:
let's store app url in the generate method and reduce passing it here


```suggestion
                $this->readmeGenerator->fillEnvironments();
```